### PR TITLE
Bug fix/skip around gather executors

### PR DESCRIPTION
**Pull request by mchacki**
This mainly adds a test framework for Gather Executors.
Also fixed a bug on the way

### CODE REVIEW

**Comment by mchacki** [comment]
Jenkins:
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/10616/

File: CHANGELOG [hunk 1]
```diff
@@ -1,6 +1,12 @@
 devel
 -----
 
+* Fixed a potential error scenario where a call could miss count skip. Have not
+  been able to write a user-query to trigger it though, so maybe this is just a
+  theoretical bug. If it hits in the number reported by skip on subsequent subqueries
+  was incorrect. First run reported correct number, second run reported again number
+  of first run, third the one of second etc.
+
```
[the remark this answers is not in the document]
Comment (mchacki) [reply CHANGELOG:9]: This PR contains the fix, as the "basic" test was red.
It is NOT the fix we have been working on in the other branch.

File: tests/Aql/GatherExecutorCommonTest.cpp [hunk 2]
```diff
@@ -0,0 +1,880 @@
...
+using namespace arangodb;
+using namespace arangodb::aql;
+
+/**
+ * @brief Description of this test case class
+ *
+ * This test class is supposed to test
+ * the data flow in gather executors.
+ * Those executors have the very special case that they have more then 1 dependency
+ * so those dependencies can be asekd in any order, and it is unclear
```
Comment (markuspf) [inline tests/Aql/GatherExecutorCommonTest.cpp:47]:
```suggestion
 * so those dependencies can be asked in any order, and it is unclear
```

**Comment by mchacki** [comment]
Jenkins after latest bugfix:

http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/10643/

Please review the last changes, the assertions did rightfully kick in in Cluster.

**Pull request merged**